web-common/src/features/entity-management/file-selectors.ts

```diff
@@ -0,0 +1,33 @@
+import { createRuntimeServiceListFiles } from "@rilldata/web-common/runtime-client";
+
+/**
+ * In dev mode we still need to get the files for left nav.
+ * This is because parse errors will not have a resource.
+ */
+export function useMainEntityFiles(
+  instanceId: string,
+  prefix: "sources" | "models" | "dashboards",
+  extension: ".yaml" | ".sql"
```

Comment:
Since `extension` can be deduced from the `prefix`, it shouldn't be a parameter of the function, rather it should be computed within the function body